### PR TITLE
Change reheat coil sizing to use air terminal unit air flow rate to size the water coil UA value.

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
The water reheat coil currently sizes the water flow rate and coil capacity based on the terminal unit air flow rate used during heating operation. The model then used the greater of the zone heating and cooling air flow rate to calculate the water coil UA. This change addresses #6044 and the model now uses the heating air flow rate to size all coil model parameters and results in a decrease to the calculated coil UA. Small differences in performance is expected for existing input files (e.g., higher water flow rate and lower water side delta T) with minor changes to annual energy use.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
Results of V8.6:

    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size Design Coil Load [W], 2564.72868
    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size Maximum Water Flow Rate [m3/s], 5.57183E-005
    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size U-Factor Times Area Value [W/K], 277.54959

versus this branch:

    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size Design Coil Load [W], 2564.72868
    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size Maximum Water Flow Rate [m3/s], 5.57183E-005
    Component Sizing Information, Coil:Heating:Water, SPACE5-1 ZONE COIL, Design Size U-Factor Times Area Value [W/K], 63.32069

The results bring back a memory of users complaining that the return water temperature is too low. However, if you size a HX at one air flow rate to calculate the water flow rate, then operate that HX at a higher air flow rate, then the outlet water temperature should be lower, yes?

    Component Sizing Information, AirTerminal:SingleDuct:ConstantVolume:Reheat, SPACE5-1 CV REHEAT, Design Size Maximum Air Flow Rate [m3/s], 0.27891
    Component Sizing Information, AirTerminal:SingleDuct:ConstantVolume:Reheat, SPACE5-1 CV REHEAT, Design Size Reheat Coil Sizing Air Volume Flow Rate [m3/s], 6.36312E-002

If the terminal unit operated at the air flow rate used to spec the coil, then the outlet water temperature would be higher.

This on hold until we decide what to do. @EnergyArchmage @mjwitte 

UPDATE: I had this backwards. The outlet water temperature is higher now since the UA was reduced. This actually makes more sense than lower UA AND colder outlet water temp.

5zoneaircooledconvcoef

**Comment by mjwitte** [comment]
@rraustad This branch has a conflict now.

**Comment by mjwitte** [comment]
And the PR needs a description at the top

**Comment by nrel-bot-3** [comment]
@rraustad @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@rraustad @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@rraustad @lgentile it has been 16 days since this pull request was last updated.

File: src/EnergyPlus/ReportSizingManager.cc
Comment (Myoldmopar) [inline src/EnergyPlus/ReportSizingManager.cc]: What's the reasoning behind using `CWInitConvTemp` for `rho` but `HWInitConvTemp` for `Cp`?  Not sure how much difference it's going to make but it looks weird.

File: src/EnergyPlus/ReportSizingManager.cc
Comment (Myoldmopar) [inline src/EnergyPlus/ReportSizingManager.cc]: Extra indent made it in here somehow.

File: src/EnergyPlus/ReportSizingManager.cc
Comment (Myoldmopar) [inline src/EnergyPlus/ReportSizingManager.cc]: Similar extra indent

File: src/EnergyPlus/ReportSizingManager.cc
Comment (Myoldmopar) [inline src/EnergyPlus/ReportSizingManager.cc]: Yeah, I mean this appears reasonable, but because of the really chunky changes to sizing coming out as a result of this change, I'll need to learn more about `DataWaterFlowUsedForSizing` before merging.  I'm afraid the change in sized values will be alarming to many users.

File: src/EnergyPlus/ReportSizingManager.cc
Comment (rraustad) [reply src/EnergyPlus/ReportSizingManager.cc]: Legacy code. These should be changed to HW now or eventually.

File: src/EnergyPlus/ReportSizingManager.cc
Comment (rraustad) [reply src/EnergyPlus/ReportSizingManager.cc]: I think the only sizing change was a lower UA. There must be some physical limit to UA. Some of the coils have what appears to be unreasonably high UA values. For that reason, I thought this change looked reasonable.

**Review by Myoldmopar — APPROVED** [review]
Code changes all make sense, building and running locally right now to spot test a few things, but marking it approved for now.

**Comment by Myoldmopar** [comment]
The merge into develop of #5851 included an adjustment on the water coil sizing.  This PR also had a small tweak.  Because of that I had to tweak this one during the merge conflict resolution to get the combined tweaked value.  All good.

I also cleaned up some stray compiler warnings.  This is good to go.  Merging.

**Pull request merged**
